test/framework/src/main/java/org/elasticsearch/cluster/coordination/DeterministicTaskQueue.java

```diff
@@ -312,7 +313,25 @@ public ExecutorService executor(String name) {
 
             @Override
             public ScheduledFuture<?> schedule(TimeValue delay, String executor, Runnable command) {
-                scheduleAt(currentTimeMillis + delay.millis(), command);
+                final AtomicInteger taskState = new AtomicInteger();
+                final int NOT_STARTED = 0;
```

Comment:
just to make it more explicit, can you initialize taskState with NOT_STARTED

Reply:
Yep, I pushed 70e4f9c6fb3dd3e57bad3bc917a1f610c26a9fbd.